### PR TITLE
Implement the read only functionality for log storage

### PR DESCRIPTION
**Pull request by Martin2112**
And add some tests to show that it works

### CODE REVIEW

File: storage/mysql/tree_storage_test.go [hunk 1]
```diff
@@ -104,6 +144,22 @@ func createTestDB() {
 	if err != nil {
 		panic(err)
 	}
+	// Create a second tree set up to be read only. Have to remove tree control row manually
+	// to avoid a constraint issue
+	_, err = db.Exec("DELETE FROM TreeControl WHERE TreeId=24")
+	if err != nil {
+		panic(err)
```
Comment (AlCutter) [inline storage/mysql/tree_storage_test.go]: All the `panic()` calls in here should probably be `t.Fatalf()`.

Comment (Martin2112) [inline storage/mysql/tree_storage_test.go]: This gets called from TestMain, which doesn't have a testing.T available.

**Comment by Martin2112** [comment]
Merge conflicts fixed up.

File: storage/mysql/tree_storage_test.go (hunk 1 above)
Comment (AlCutter) [inline storage/mysql/tree_storage_test.go]: Rats.

File: storage/mysql/tree_storage.go [hunk 2]
```diff
@@ -102,8 +104,17 @@ func NewLogStorage(id trillian.LogID, url string) (storage.LogStorage, error) {
 		return nil, err
 	}
 
+	err = db.QueryRow(getTreeParametersSql, id.TreeID).Scan(&readOnlyRequests)
+
+	// TODO: It's probably not ok for the log to have no prameters set but we haven't got an
```
Comment (AlCutter) [inline storage/mysql/tree_storage.go]: "prameters"

File: storage/mysql/tree_storage.go [hunk 3]
```diff
@@ -102,8 +104,17 @@ func NewLogStorage(id trillian.LogID, url string) (storage.LogStorage, error) {
...
+	// admin API yet
+	if err == sql.ErrNoRows {
+		readOnlyRequests = false
```
Comment (AlCutter) [inline storage/mysql/tree_storage.go]: big scary `log.Warning(...)` and a `TODO` to remove this temporary work-around?

Comment (Martin2112) [inline storage/mysql/tree_storage.go]: Done.

File: storage/mysql/tree_storage.go (hunk 2 above)
Comment (Martin2112) [inline storage/mysql/tree_storage.go]: Done.

**Comment by Martin2112** [comment]
Ping!

**Comment by Martin2112** [comment]
Underlying storage has diverged too much. Will be quicker to recreate this PR against current master.

**Comment by Martin2112** [comment]
Obsoleted by #40.

**Pull request closed without merging**